File: plugins/scandit-sdk/evals/matrixscan-count-android/fixtures/IntegratedActivity.kt
package com.example.myapp

import android.os.Bundle
import android.widget.FrameLayout
import androidx.appcompat.app.AppCompatActivity
import com.scandit.datacapture.barcode.count.capture.BarcodeCount
import com.scandit.datacapture.barcode.count.capture.BarcodeCountListener
import com.scandit.datacapture.barcode.count.capture.BarcodeCountSession
import com.scandit.datacapture.barcode.count.capture.BarcodeCountSessionSnapshot
import com.scandit.datacapture.barcode.count.capture.BarcodeCountSettings
import com.scandit.datacapture.barcode.count.ui.view.BarcodeCountView
import com.scandit.datacapture.barcode.count.ui.view.BarcodeCountViewUiListener
import com.scandit.datacapture.barcode.data.Barcode
import com.scandit.datacapture.barcode.data.Symbology
import com.scandit.datacapture.core.capture.DataCaptureContext
import com.scandit.datacapture.core.data.FrameData
import com.scandit.datacapture.core.source.Camera
import com.scandit.datacapture.core.source.FrameSourceState

class CountActivity : AppCompatActivity(), BarcodeCountListener, BarcodeCountViewUiListener {

    private val dataCaptureContext =
        DataCaptureContext.forLicenseKey("-- ENTER YOUR SCANDIT LICENSE KEY HERE --")

    private var camera: Camera? = null
    private lateinit var barcodeCount: BarcodeCount
    private lateinit var barcodeCountView: BarcodeCountView

    private var allRecognizedBarcodes: List<Barcode> = emptyList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setupRecognition()
    }

    private fun setupRecognition() {
        val cameraSettings = BarcodeCount.createRecommendedCameraSettings()
        camera = Camera.getDefaultCamera(cameraSettings)
        dataCaptureContext.setFrameSource(camera)

        val settings = BarcodeCountSettings()
        settings.enableSymbologies(
            setOf(
                Symbology.EAN13_UPCA,
                Symbology.EAN8,
                Symbology.UPCE,
                Symbology.CODE128,
                Symbology.CODE39
            )
        )

        barcodeCount = BarcodeCount.forDataCaptureContext(dataCaptureContext, settings)
        barcodeCount.addListener(this)

        val container = FrameLayout(this)
        setContentView(container)
        barcodeCountView = BarcodeCountView.newInstance(this, dataCaptureContext, barcodeCount)
        container.addView(barcodeCountView)
        barcodeCountView.uiListener = this
    }

    override fun onResume() {
        super.onResume()
        camera?.switchToDesiredState(FrameSourceState.ON)
    }

    override fun onPause() {
        super.onPause()
        camera?.switchToDesiredState(FrameSourceState.OFF)
    }

    override fun onDestroy() {
        super.onDestroy()
        barcodeCount.removeListener(this)
    }

    override fun onScan(
        barcodeCount: BarcodeCount,
        session: BarcodeCountSession,
        data: FrameData
    ) {
        val recognizedBarcodes = session.recognizedBarcodes
        runOnUiThread {
            allRecognizedBarcodes = recognizedBarcodes
        }
    }

    override fun onListButtonTapped(view: BarcodeCountView, snapshot: BarcodeCountSessionSnapshot?) {
        // Present the current progress, e.g. from snapshot?.recognizedBarcodes.
    }

    override fun onExitButtonTapped(view: BarcodeCountView, snapshot: BarcodeCountSessionSnapshot?) {
        // Present a summary, e.g. from snapshot?.recognizedBarcodes.
    }
}

File: plugins/scandit-sdk/evals/matrixscan-count-net-android/fixtures/IntegratedActivity.cs
using Android.OS;
using Android.Widget;

using Scandit.DataCapture.Barcode.Count.Capture;
using Scandit.DataCapture.Barcode.Count.UI;
using Scandit.DataCapture.Barcode.Data;
using Scandit.DataCapture.Core.Capture;
using Scandit.DataCapture.Core.Source;

namespace MyApp;

[Activity(Label = "@string/app_name", MainLauncher = true, Theme = "@style/Theme.AppCompat.Light.NoActionBar")]
public class MainActivity : CameraPermissionActivity
{
    public const string SCANDIT_LICENSE_KEY = "-- ENTER YOUR SCANDIT LICENSE KEY HERE --";

    private DataCaptureContext dataCaptureContext = null!;
    private Camera? camera;
    private BarcodeCount barcodeCount = null!;
    private BarcodeCountView barcodeCountView = null!;

    private readonly List<Barcode> scannedBarcodes = new();

    protected override async void OnCreate(Bundle? savedInstanceState)
    {
        base.OnCreate(savedInstanceState);

        this.dataCaptureContext = DataCaptureContext.ForLicenseKey(SCANDIT_LICENSE_KEY);

        this.camera = Camera.GetDefaultCamera(BarcodeCount.RecommendedCameraSettings);
        if (this.camera is not null)
        {
            await this.dataCaptureContext.SetFrameSourceAsync(this.camera);
        }

        BarcodeCountSettings settings = new BarcodeCountSettings();
        settings.EnableSymbologies(new HashSet<Symbology>
        {
            Symbology.Ean13Upca,
            Symbology.Code128,
        });
        this.barcodeCount = BarcodeCount.Create(this.dataCaptureContext, settings);

        var container = new FrameLayout(this);
        this.SetContentView(container);
        this.barcodeCountView = BarcodeCountView.Create(
            this, this.dataCaptureContext, this.barcodeCount, BarcodeCountViewStyle.Icon);
        container.AddView(this.barcodeCountView);

        this.barcodeCountView.ListButtonTapped += (s, e) => this.ShowResults();
        this.barcodeCountView.ExitButtonTapped += (s, e) => this.ShowResults();
    }

    protected override void OnResume()
    {
        base.OnResume();
        this.barcodeCount.Scanned += this.OnBarcodeCountScanned;
        this.barcodeCount.Enabled = true;
        this.RequestCameraPermission();
    }

    protected override void OnPause()
    {
        base.OnPause();
        this.camera?.SwitchToDesiredStateAsync(FrameSourceState.Off);
        this.barcodeCount.Scanned -= this.OnBarcodeCountScanned;
    }

    protected override void OnCameraPermissionGranted()
    {
        this.camera?.SwitchToDesiredStateAsync(FrameSourceState.On);
    }

    protected override void OnDestroy()
    {
        base.OnDestroy();
        this.barcodeCount.Scanned -= this.OnBarcodeCountScanned;
        this.barcodeCountView.Dispose();
        this.barcodeCount.Dispose();
    }

    private void OnBarcodeCountScanned(object? sender, BarcodeCountEventArgs args)
    {
        List<Barcode> recognized = args.Session.RecognizedBarcodes.ToList();
        this.RunOnUiThread(() =>
        {
            this.scannedBarcodes.Clear();
            this.scannedBarcodes.AddRange(recognized);
        });
    }

    private void ShowResults()
    {
        // Present this.scannedBarcodes to the user.
    }
}
